packages/status-js/src/client/chat.ts

```diff
@@ -0,0 +1,409 @@
+import { hexToBytes } from 'ethereum-cryptography/utils'
+import { PageDirection } from 'js-waku'
+
+import { ChatMessage as ChatMessageProto } from '~/protos/chat-message'
+import { CommunityRequestToJoin } from '~/protos/communities'
+import { EmojiReaction } from '~/protos/emoji-reaction'
+
+import { idToContentTopic } from '../contentTopic'
+import { createSymKeyFromPassword } from '../encryption'
+import { getReactions } from './community/get-reactions'
+
+import type { MessageType } from '../../protos/enums'
+import type { Client } from '../client'
+import type { Community } from './community/community'
+import type { Reactions } from './community/get-reactions'
+import type { ImageMessage } from '~/src/proto/communities/v1/chat_message'
+import type { CommunityChat } from '~/src/proto/communities/v1/communities'
+import type { WakuMessage } from 'js-waku'
+
+export type ChatMessage = ChatMessageProto & {
+  messageId: string
+  pinned: boolean
+  reactions: Reactions
+  chatUuid: string
+  responseToMessage?: Omit<ChatMessage, 'responseToMessage'>
+}
+
+export class Chat {
+  private readonly community: Community
+  private readonly client: Client
+
+  public readonly uuid: string
+  public readonly id: string
+  public readonly contentTopic: string
+  public readonly type: MessageType.COMMUNITY_CHAT
+  public readonly symetricKey: Uint8Array
+  public description: CommunityChat
+  public readonly chatCallbacks: Set<(description: CommunityChat) => void>
+  public messages: ChatMessage[]
+  public readonly messageCallbacks: Set<(messages: ChatMessage[]) => void>
+
+  constructor(
+    community: Community,
+    client: Client,
+    uuid: string,
+    id: string,
+    contentTopic: string,
+    type: MessageType.COMMUNITY_CHAT,
+    symetricKey: Uint8Array,
+    description: CommunityChat
+  ) {
+    this.client = client
+    this.community = community
+
+    this.uuid = uuid
+    this.id = id
+    this.contentTopic = contentTopic
+    this.type = type
+    this.symetricKey = symetricKey
+    this.description = description
+
+    this.chatCallbacks = new Set()
+    this.messages = []
+    this.messageCallbacks = new Set()
+  }
+
+  public static create = async (
+    community: Community,
+    client: Client,
+    uuid: string,
+    type: MessageType.COMMUNITY_CHAT,
+    description: CommunityChat
+  ) => {
+    const id = `${community.publicKey}${uuid}`
+    const contentTopic = idToContentTopic(id)
+    const symetricKey = await createSymKeyFromPassword(id)
+
+    return new Chat(
+      community,
+      client,
+      uuid,
+      id,
+      contentTopic,
+      type,
+      symetricKey,
+      description
+    )
+  }
```

Comment:
There is no need for a static function when this can be done in the `Chat` constructor.